gemma-core/src/test/java/ubic/gemma/persistence/service/analysis/expression/diff/DifferentialExpressionAnalysisDaoTest.java

```diff
@@ -61,13 +62,18 @@ public void testCreateAnalysisWithResultSetAndPvalueDistribution() {
         analysis = differentialExpressionAnalysisDao.create( analysis );
         sessionFactory.getCurrentSession().flush();
         assertNotNull( analysis.getId() );
+        assertEquals( 3, analysis.getResultSets().size() );
         for ( ExpressionAnalysisResultSet resultSet : analysis.getResultSets() ) {
             assertNotNull( resultSet.getId() );
             assertNotNull( resultSet.getPvalueDistribution().getId() );
+            assertEquals( 100, resultSet.getResults().size() );
             for ( DifferentialExpressionAnalysisResult result : resultSet.getResults() ) {
                 assertNotNull( result.getId() );
+                assertFalse( sessionFactory.getCurrentSession().contains( result ) );
+                assertEquals( 2, result.getContrasts().size() );
                 for ( ContrastResult contrast : result.getContrasts() ) {
                     assertNotNull( contrast.getId() );
+                    assertFalse( sessionFactory.getCurrentSession().contains( contrast ) );
```

Comment:
It would be nice to make it so that the manually inserted contrast and result are made part of the session. However since they are immutable, it's not worrisome.